src/ethereum/EthereumTools.ts

```diff
@@ -381,6 +382,69 @@ export class EthereumTools implements EdgeCurrencyTools {
     return Object.keys(ethereumPlugins).map(plugin => `wallet:${plugin}`)
   }
 
+  async getTokenDetails(filter: EdgeGetTokenDetails): Promise<EdgeToken[]> {
+    const { contractAddress } = filter
+    if (contractAddress == null) return []
+
+    const valid = EthereumUtil.isValidAddress(contractAddress)
+    if (!valid) return []
+
+    const { networkAdapterConfigs } = this.networkInfo
+
+    const networkAdapterConfig = networkAdapterConfigs.find(
+      (networkAdapterConfig): networkAdapterConfig is RpcAdapterConfig =>
+        networkAdapterConfig.type === 'rpc'
+    )
+
+    if (networkAdapterConfig == null)
+      throw new Error('getTokenDetails: No RpcAdapterConfig')
+
+    const rpcServers = networkAdapterConfig.servers
+
+    const ethProviders: ethers.providers.JsonRpcProvider[] = rpcServers.map(
+      rpcServer =>
+        new ethers.providers.JsonRpcProvider(
+          rpcServer,
+          this.networkInfo.chainParams.chainId
+        )
+    )
+    const ERC20_ABI = [
+      'function name() view returns (string)',
+      'function symbol() view returns (string)',
+      'function decimals() view returns (uint8)'
+    ]
+
+    const { name, symbol, decimals } = await multicastEthProviders<
+      { name: string; symbol: string; decimals: number },
+      ethers.providers.JsonRpcProvider
+    >({
+      func: async (ethProvider: ethers.providers.JsonRpcProvider) => {
+        const contract = new ethers.Contract(
+          contractAddress,
+          ERC20_ABI,
+          ethProvider
+        )
+        const [name, symbol, decimals] = await Promise.all([
+          contract.name(),
+          contract.symbol(),
+          contract.decimals()
+        ])
+        return { name, symbol, decimals }
+      },
+      providers: ethProviders
+    })
```

Comment:
You could do `.catch(() => [])`.

Otherwise, presumably the GUI needs a `.catch(() => [])` on the token lookup. Maybe that is the way, in which we can skip the change.